gateway/coprocess_test.go

```diff
@@ -137,3 +139,103 @@ func Test_shouldAddConfigData(t *testing.T) {
 		})
 	}
 }
+
+func TestSyncHeadersAndMultiValueHeaders(t *testing.T) {
+	// defining the test cases
+	testCases := []struct {
+		name                      string
+		headers                   map[string]string
+		initialMultiValueHeaders  []*coprocess.Header
+		expectedMultiValueHeaders []*coprocess.Header
+	}{
+		{
+			name: "adding a header",
+			headers: map[string]string{
+				"Header1": "value1",
+				"Header2": "value2",
+			},
+			initialMultiValueHeaders: []*coprocess.Header{
+				{
+					Key:    "Header1",
+					Values: []string{"oldValue1"},
+				},
+			},
+			expectedMultiValueHeaders: []*coprocess.Header{
+				{
+					Key:    "Header1",
+					Values: []string{"value1"},
+				},
+				{
+					Key:    "Header2",
+					Values: []string{"value2"},
+				},
+			},
+		},
+		{
+			name: "removing a header",
```

Comment:
it cannot be done, basically because our code doesn't remove headers it just receive 2 structures that currently store headers: header (map of string-string) and multivalueheaders that is a map of string key and value an array of strings, so, it know the status before executing the plugin middleware, then the plugin can do anything: remove, add or update headers and then the gateway reads the result of that operation in which the header or headers could have changed, so, at this point we synchronize the keys, we basically check what was removed, added or updated from the univalue struct and replicate those changes in the multivalue struct. So, we dont remove headers, we just replicate changes